### PR TITLE
Improve logging using Zap library

### PR DESCRIPTION
**Pull request by abusarah-tech**

### CODE REVIEW

File: src/api/apiServer.go [hunk 1]
```diff
@@ -2,44 +2,45 @@ package api
...
-		fmt.Printf("Info - HTTP about to listen on %s.", port)
-		log.Printf("Info - HTTP about to listen on %s.", port)
+		combinedLogger.Info(fmt.Sprintf("HTTP about to listen on %s.", port))
 
 		resolve, _ := net.ResolveTCPAddr("tcp4", "0.0.0.0:"+port)
 		resolveTCP, _ := net.ListenTCP("tcp4", resolve)
 
 		errServer := http.Serve(resolveTCP, newRouter)
-		log.Fatal("Error - Startup of API server", errServer)
+		combinedLogger.Error(fmt.Sprintf("Failed to startup of API server %s", errServer))
```
Comment (RaspberryTech01) [inline src/api/apiServer.go]: Might be a better idea to not use `fmt.Sprintf` and just call the error as

``combinedLogger.Error("Parsing request", err)``
or
``combinedLogger.Error("Parsing request", err.Error())``

Comment (abusarah-tech) [reply src/api/apiServer.go]: `zap.Logger` does not support `... interface{}` at the second argument. We keep it that way or we build a custom type logger that wrap around the lib and call `Sprintf` only in that file.

Comment (RaspberryTech01) [reply src/api/apiServer.go]: What about instead of `"error here", error.Error()`

We use `"error here" + error.Error()`?

Comment (abusarah-tech) [reply src/api/apiServer.go]: That does not solve it for certain logged informations like integers and other types.

**Review by RaspberryTech01 — APPROVED** [review]
Thanks mate

**Pull request merged**